src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -85,6 +85,22 @@ internal sealed partial class DocumentProvider : ForegroundThreadAffinitizedObje
             Marshal.ThrowExceptionForHR(runningDocumentTableForEvents.AdviseRunningDocTableEvents(new RunningDocTableEventsSink(this), out _runningDocumentTableEventCookie));
         }
 
+        [Obsolete("Use the overload with 'isAdditionalFile' instead", error: true)]
+        public IVisualStudioHostDocument TryGetDocumentForFile(
+            IVisualStudioHostProject hostProject,
+            string filePath,
+            SourceCodeKind sourceCodeKind,
+            Func<ITextBuffer, bool> canUseTextBuffer,
+            Func<uint, IReadOnlyList<string>> getFolderNames,
+            EventHandler updatedOnDiskHandler = null,
+            EventHandler<bool> openedHandler = null,
+            EventHandler<bool> closingHandler = null)
+        {
+            return TryGetDocumentForFile(
+                hostProject, filePath, sourceCodeKind, canUseTextBuffer, getFolderNames, isAdditionalFile: false,
+                updatedOnDiskHandler: updatedOnDiskHandler, openedHandler: openedHandler, closingHandler: closingHandler);
+        }
```

Comment:
Is there a bug filed to remove this later? Having internal code peppered with Obsolete messages like this because of IVT makes it even scarier to make *any* change in Roslyn for fear of breaking a caller.